site/blog/tutorials/iasql-on-ec2.mdx

```diff
@@ -0,0 +1,284 @@
+```sql
+INSERT INTO instance_security_groups (instance_id, security_group_id)
+SELECT (SELECT id
+        FROM instance
+        WHERE tags ->> 'name' = 'iasql'
+        LIMIT 1),
+       (SELECT id
+        FROM security_group
+        WHERE group_name = 'iasql_sg'
+        LIMIT 1);
+```
+The `instace_security_groups` table is a Many-to-Many join table that connects `instance`s to their `security_group`s. Now we need to wait for IaSQL to start syncing the state of the cloud with our local changes. This will normally happen every 2 minutes. There is a way to make the cloud apply/sync happen immediately using [IaSQL transactions](https://iasql.com/docs/transaction/), but for the purpose of this post we'll wait for the next cronjob tick to apply our changes.
+
+After waiting long enough for the server to start and for the startup script to finish, we can use instance's public IP to access our self-hosted IaSQL dashboard:
+
+```sql
+SELECT public_ip_address
+FROM instance_metadata
+WHERE id = (SELECT id
+            FROM instance
+            WHERE tags ->> 'name' = 'iasql'
+            LIMIT 1);
+```
+
+Now we can go to `http://<instance-ip>:9876/` URL, and we'll see the beautiful IaSQL dashboard deployed on our EC2 server.
+
+##### Optional: Cloud-init Directives
+Now let's use cloud-init directives to do the same thing. This part is not exactly related to IaSQL, or what we intended to do in this post. It's just an exploration for other choices we have with EC2 user data field.
+
+For the previous part, we used the following bash script as the `user_data` column in the `instance` table:
+```bash
+#!/bin/bash
+sudo apt-get upgrade -y
+curl -fsSL https://get.docker.com -o get-docker.sh
+sudo sh get-docker.sh
+docker run -p 9876:9876 -p 5432:5432 --name iasql iasql/iasql
+```
+The `user_data` field [can directly take](https://cloudinit.readthedocs.io/en/latest/explanation/format.html#user-data-script) bash scripts, as above, and also it can contain [cloud-init](https://cloud-init.io/) directives. Let's first break the above script down to see what exactly it does:
+- `sudo apt-get upgrade -y`: used to install the newest versions of all packages currently installed on the system
+- `curl -fsSL https://get.docker.com -o get-docker.sh`: download the Docker install script, then it's executed by `sudo sh get-docker.sh`
+- `docker run -p 9876:9876 -p 5432:5432 --name iasql iasql/iasql`: run the IaSQL container
+Knowing the intent of each command, it's not hard to do the same functionality using `cloud-init`.
+```yaml
+#cloud-config
+package_update: true # apt-get update - this was in the `get-docker.sh` script so we didn't manually run it in our script
+package_upgrade: true # apt-get upgrade
+apt:
+  sources:
+    docker.list: # add the docker source for the package manager
+      source: deb [arch=amd64] https://download.docker.com/linux/ubuntu $RELEASE stable
+      keyid: 9DC858229FC7DD38854AE2D88D81803C0EBFCD88
```

Comment:
This section here undermines the narrative below because it is still `apt`-specific logic.

However, I think it is salvageable with some other changes.